### DISCUSSION TITLE
Return `stock` in cart product data

### DISCUSSION THREAD

**Discussion by el-schneider** [duncanmcclean/simple-commerce#798] [Feature Requests]
Currently this is what I get from the items array, when updating the cart and it would be helpful to me if `product` also would contain `stock`.

```json
[
    {
        "id": "fac03902-f94b-4296-ba5c-afab1637a4ff",
        "product": {
            "id": "a979ab16-04b0-4194-a08d-a7c7b3d7ca5c",
            "title": "Some Product",
            "url": "/products/some-product",
            "permalink": "http://some.domain/products/some-product",
            "api_url": null
        },
        "variant": null,
        "quantity": 2,
        "total": "€70.59",
        "metadata": [],
        "tax": {
            "amount": "€13.41",
            "rate": 19,
            "price_includes_tax": true
        }
    },
]
```

**Comment by el-schneider** [comment]
I just realized, that the data returned by the different actions is not always the same. Sometimes it get's augmented and sometimes not. I wonder if it would be feasable to always augment it, as it would help me a lot. E.g. `CartController::index()` returns and augmented array while `CartControllerItem::update()` does not, hence one has `stock` and one does not.

**Comment by duncanmcclean** [comment]
In your example, Simple Commerce is returning an [`EntryResource`](https://github.com/statamic/cms/blob/3.4/src/Http/Resources/API/EntryResource.php#L15) which 'shallow augments' the fields.

If you extend Statamic's `Entry` class, you may specify what fields on entries are returned by 'shallow augmentation', [review the Statamic documentation](https://statamic.dev/extending/augmentation#shallow-augmentation) for more information.

I'm not sure why one controller would be returning a fully augmented version of fields and the other not because they're both using `->toResource()`:

* https://github.com/duncanmcclean/simple-commerce/blob/main/src/Http/Controllers/CartController.php#L136
* https://github.com/duncanmcclean/simple-commerce/blob/main/src/Http/Controllers/CartItemController.php#L222

In the future, we could potentially look at extending what's returned by the actions to stuff that's more helpful. However, that would likely need to wait until a breaking release, like v5.0.

**Reply by duncanmcclean** [reply]
I've added "making actions return augmented versions of orders" to my v5.0 checklist. I don't have a date for when this will be released though.

**Reply by duncanmcclean** [reply]
In case anyone wants to do this, here's some instructions on how to override the `Entry` class to return the `price` field on products in the line items array: https://gist.github.com/duncanmcclean/f68c25b58db5141f9c7ca9d09cf4a25f

**Comment by el-schneider** [comment]
I think this is where it is getting augmented: https://github.com/duncanmcclean/simple-commerce/blob/46c4e8873eb8f22ddb180c4f101301affcb200c4/src/Http/Controllers/CartController.php#L26

I'll look into the docs.

**Reply by duncanmcclean** [reply]
Ah yes, sorry, the `index` action is augmented. I must have read your previous comment as the `store` method 🙈

**Reply by duncanmcclean** [reply]
As we can't change the other action endpoints to being augmented until a breaking version, I'd recommend pinging that `index` cart endpoint anytime you need something that's not otherwise provided.
